client/src/GraphQLFileUploader.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Libero\SharedServicesExperiment\Client;
+
+use InvalidArgumentException;
+use Softonic\GraphQL\Client as GraphQLClient;
+use Softonic\GraphQL\ResponseBuilder;
+use GuzzleHttp\ClientInterface;
+use RuntimeException;
+
+use function GuzzleHttp\Psr7\stream_for;
+
+class GraphQLFileUploader implements FileUploader
+{
+    /**
+     * GraphQL client instance
+     *
+     * @var GraphQLClient
+     */
+    private $client;
+
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = new GraphQLClient($client, new ResponseBuilder());
+    }
+
+    public function uploadFile(string $sourcePath, string $uploadPath): FileUploadRecord
+    {
+        if (! file_exists($sourcePath)) {
+            throw new InvalidArgumentException('File not found: ' . $sourcePath);
+        }
+
+        $finfo         = finfo_open(FILEINFO_MIME_TYPE);
+        $mimeType      = finfo_file($finfo, $sourcePath);
+        $contentLength = filesize($sourcePath);
+        $body          = stream_for(fopen($sourcePath, 'r'));
+        list(, $path)  = explode('/', $uploadPath, 2) + [1 => null];
+
+        $query = <<<'QUERY'
```

Comment:
is this supposed a schema that is supposed to be hardcoded in the client?

Reply:
it defines the which mutations to use on the server side. So here 'UploadFile' uses the 'uploadFile' mutation (which is resolved in the storage service)